python/plugins/processing/algs/qgis/CreateConstantRaster.py

```diff
@@ -61,28 +76,35 @@ def displayName(self):
         return self.tr('Create constant raster layer')
 
     def processAlgorithm(self, parameters, context, feedback):
-        layer = self.parameterAsRasterLayer(parameters, self.INPUT, context)
+        extent = self.parameterAsExtent(parameters, self.EXTENT, context)
+        crs = self.parameterAsCrs(parameters, self.TARGET_CRS, context)
         value = self.parameterAsDouble(parameters, self.NUMBER, context)
+        pixelSize = self.parameterAsDouble(parameters, self.PIXEL_SIZE, context)
 
         outputFile = self.parameterAsOutputLayer(parameters, self.OUTPUT, context)
+        outputFormat = raster.formatShortNameFromFileName(outputFile)
```

Comment:
What do you feel about porting raster.formatShortNameFromFileName to c++ as QgsRasterFileWriter::driverForExtension() ? We have QgsVectorFileWriter::driverForExtension() already, so it would be nice to also have a single reusable c++ method for doing the same with raster formats. The code should be a direct copy and paste from https://github.com/qgis/QGIS/commit/ce4f93b4101c8518ac036a5cabdb74f0fda003d0 and https://github.com/qgis/QGIS/commit/03dd26ef9fe38f59eac699e7f215efd5b94af895 but changing GDAL_DCAP_VECTOR for GDAL_DCAP_RASTER